src/olympia/blocklist/tasks.py

```diff
@@ -161,7 +161,16 @@ def upload_filter(generation_time, filter_list=None, create_stash=False):
             # If we have a stash, write that
             stash_upload_data = {
                 'key_format': MLBF.KEY_FORMAT,
-                'stash_time': generation_time,
+                # We subtract 1 from the stash time to ensure that FX
+                # considers all stashes to be older than base filters
+                # generated at the same time. The filter timestamp is
+                # tied to the MLBF creation time so we cannot change it.
+                # FX checks stashes before filters and only if a filter
+                # is more recent than the stash. So making the stash time
+                # 1 ms older than the filter ensures if an item is added
+                # via a new filter, and a stash was also created at the same
+                # time, the item will be found in the filter.
+                'stash_time': generation_time - 1,
```

Comment:
Let's reserve that kind of thing for https://github.com/mozilla/addons/issues/15200